modules/ding_place2book/ding_place2book.fields.inc

```diff
@@ -414,13 +414,20 @@ function ding_place2book_field_presave($entity_type, $entity, $field, $instance,
     case 'ding_p2b':
       if (!empty($items)) {
         $settings = reset($items);
+        // Boolean indicating if the p2b synchronizationn was successful.
+        $synchronized = FALSE;
         // Ensure that both 'synchronize' and 'do_sync' is set and TRUE, before
         // making any requests to p2b API. This is to ensure we don't make lots
         // of unneeded requests when nodes are saved programmatically or updated
         // in bulk via the UI.
         // See ding_place2book_field_widget_form() for more info.
         if ($settings['synchronize'] && !empty($settings['do_sync'])) {
-          $items = _ding_place2book_sync_p2b_entities($entity, $settings);
+          $synchronized = _ding_place2book_sync_p2b_entities($entity, $settings);
```

Comment:
This seems odd to me. According to the DocBlock `_ding_place2book_sync_p2b_entities` returns an array. This change seems to assume that it can be a boolean. Does the DocBlock need updating or are we handling a case where we might get an empty array returned in case on an error?

Could we simplify the change to something like:

```php
$synchronized_items = _ding_place2book_sync_p2b_entities($entity, $settings);
// Only update field data if synchronization was successful.
if (!empty($synchronized_items)) {
  $items = $synchronized_items;
}
```

Reply:
With your suggestion we don't have to change return value and update docblock, so I think we should go with that. I'll update PR.